packages/termux-tools/pkg

```diff
@@ -18,8 +18,50 @@ show_help() {
 	exit 1
 }
 
-check_pkgcache() {
-	if [ -z "$(find @TERMUX_PREFIX@/var/cache/apt/pkgcache.bin -mmin -5)" ]; then
+select_mirror() {
+	local main_repo="https://dl.bintray.com/termux/termux-packages-24"
+	local top_mirrors="${main_repo} https://termux.mentality.rip/termux-packages-24 https://grimler.se/termux-packages-24 https://main.termux-mirror.ml"
+
+	local current_mirror
+	current_mirror=$(grep -P "^\s*deb\s+" @TERMUX_PREFIX@/etc/apt/sources.list | grep -oP 'https?://[a-z0-9/._-]+')
+
+	# Do not update mirror if:
+	# * Uses .cn domain - specific to Chinese users.
+	# * If $TERMUX_PKG_NO_MIRROR_SELECT was set.
+	if [ -n "${TERMUX_PKG_NO_MIRROR_SELECT-}" ] || grep -qP '.+\.cn/' <(echo "${current_mirror-x}"); then
+		return
+	fi
+
+	if [ -n "$(find @TERMUX_PREFIX@/var/cache/apt/pkgcache.bin -mmin -30 2>/dev/null)" ]; then
+		# Mirrors are rotated if either they are not working or 30 minutes timeout
+		# has passed.
+		if [ -n "${current_mirror}" ] && curl --head --fail --location "${current_mirror}/dists/stable/Release" >/dev/null 2>&1; then
+			echo "Reusing mirror: ${current_mirror}"
+			return
+		fi
+	fi
+
+	# Shuffle mirrors and pick the first one which is accessible.
+	local selected_mirror m
+	for m in $(echo "$top_mirrors" | tr ' ' '\n' | shuf); do
+		echo "Trying mirror: $m"
+		if curl --head --fail --location "${m}/dists/stable/Release" >/dev/null 2>&1; then
+			selected_mirror="$m"
+			break
+		else
+			continue
+		fi
+
+		selected_mirror="${main_repo}"
+		echo "Using fallback mirror: ${main_repo}"
+		break
+	done
+
+	echo "deb ${selected_mirror}/ stable main" > @TERMUX_PREFIX@/etc/apt/sources.list
```

Comment:
If a user searches for some package once an hour, this will write a new rotated mirror to the file every hour. Do we want to rotate it that much? :stuck_out_tongue_winking_eye:

Reply:
I can increase it, say to 3 hours.

Reply:
Done.

Reply:
I think 10 would be fine, but I don't want to bikeshed this, so I'll let you decide what's best.